soc/renesas/smartbond/da1469x/power.c

```diff
@@ -22,6 +22,18 @@ static int pdc_idx_combo;
 static int pdc_idx_sw_trigger;
 static bool wait_for_jtag;
 
+#define GPIO0_NGPIOS DT_PROP(DT_NODELABEL(gpio0), ngpios)
+#define GPIO1_NGPIOS DT_PROP(DT_NODELABEL(gpio1), ngpios)
+
+struct gpio_state_data {
+	uint32_t data0;
+	uint32_t mode0[GPIO0_NGPIOS];
+	uint32_t data1;
+	uint32_t mode1[GPIO1_NGPIOS];
+};
+
```

Comment:
again, stuff that should be in driver/ code in soc/. Please, find a better solution. If Zephyr notification mechanism isn't sufficient, you can improve it or look at suggested improvements that are on PR.

Reply:
latching gpio moved to gpio driver